drift/lib/src/runtime/types/converters.dart

```diff
@@ -35,25 +35,30 @@ abstract class TypeConverter<D, S> {
 /// `.drift` files) refers to a type converter that inherits from
 /// [JsonTypeConverter], it will also be used for the conversion from and to
 /// JSON.
-mixin JsonTypeConverter<D, S> on TypeConverter<D, S> {
+abstract class JsonTypeConverter<D, S, J> extends TypeConverter<D, S> {
```

Comment:
One way to make this non-breaking might be to introduce this as a new superclass (imagine `JsonTypeConverterWithDifferentTypes` but with a less cumbersome name).

Them, `typealias JsonTypeConverter<D, S> = JsonTypeConverterWithDifferentTypes<D, S, S>;` would make this non-breaking I believe? Or just `class JsonTypeConverter<D, S> extends JsonTypeConverterWithDifferentTypes<D, S, S>`. I might still be overlooking something, but maybe that works to make this non-breaking.

Reply:
I'll give this a go later

Reply:
I think it's backwards compatible now. Thanks for the suggestion!

Not sure what to name `JsonTypeConverterWithDifferentTypes`, it seems a little verbose.

Reply:
Nice!